### PR TITLE
Console compiler: exit with error status for serious warnings

### PR DESCRIPTION
**Pull request by cgay**
Console compiler: exit with error status for serious warnings

Issue #1373

### CODE REVIEW

**Comment by housel** [comment]
Why not add a command-line option to suppress this new exit status behavior, which could then be used when building libraries that deliberately induce serious warnings (such as the common-dylan-test-suite)?

**Comment by cgay** [comment]
I was planning to add that flag in a later commit (please comment on the bug if you have preferences for what it should look like). 

I would like to be able to detect _new_ serious warnings in the test suite and Open Dylan code, which is why I prefer to comment out those two tests. I could make the exit code be the total number of serious warnings (maxing out at 127, I guess, if that's a shell limitation...127 should be enough serious warnings for anyone) and then callers who expect non-zero could compare with that?

File: sources/environment/commands/command-line.dylan [hunk 1]
```diff
@@ -8,6 +8,11 @@ Warranty:  Distributed WITHOUT WARRANTY OF ANY KIND
 
 /// Useful constants
 
+define constant $success-exit-code          = 0;
+define constant $unexpected-error-exit-code = 1;
+define constant $usage-error-exit-code      = 2;
```
Comment (housel) [inline sources/environment/commands/command-line.dylan:13]: Maybe sort these by seriousness so they could be used with `IF ERRORLEVEL` https://devblogs.microsoft.com/oldnewthing/20080926-00/?p=20743

**Review by housel — APPROVED** [review]

File: sources/environment/commands/command-line.dylan (hunk 1 above)
Comment (cgay) [reply sources/environment/commands/command-line.dylan]: Done

**Pull request merged**